lib/Doctrine/ORM/Persisters/Collection/OneToManyPersister.php

```diff
@@ -182,10 +182,15 @@ public function removeElement(PersistentCollection $collection, $element)
             return false;
         }
 
+        $this->uow->scheduleOrphanRemoval($element);
+
+        return true;
+        /*
         return $this
```

Comment:
OrphanRemoval its hook for remove Entity from collection and it doesn't matter which it type of.
If no OrphanRemoval we do 2 operation: remove from collection and remove from EM, if yes - just one remove from collection and Doctrine automatically collect entity for deletion.
And I think this operation should be execute in transaction on flush. Early, I saw issue about this.